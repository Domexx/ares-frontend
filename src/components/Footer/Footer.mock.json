[
  {
    "headline": "Support",
    "children": [
      {
        "name": "Passwort vergessen?",
        "link": "https://nowhere.de"
      },
      {
        "name": "Hilfe Center",
        "link": "https://nowhere.de"
      },
      {
        "name": "Entwickler-Dokumentation",
        "link": "https://nowhere.de"
      },
      {
        "name": "Kontaktformular",
        "link": "https://nowhere.de"
      }
    ]
  },
  {
    "headline": "Informationen",
    "children": [
      {
        "name": "Was ist Habbo?",
        "link": "https://nowhere.de"
      },
      {
        "name": "Wie wird gespielt?",
        "link": "https://nowhere.de"
      },
      {
        "name": "Habbo Benimmregeln",
        "link": "https://nowhere.de"
      },
      {
        "name": "Sicherheit",
        "link": "https://nowhere.de"
      }
    ]
  }
]
