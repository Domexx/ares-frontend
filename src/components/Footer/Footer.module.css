/* === === === === Footer === === === === */

.Footer {
  position: relative;
  flex: none;
  width: 100%;
  max-height: 435px;
}

.FooterTop {
  grid-column: 1 / -1;

  padding: 40px 0;
  background: var(--clr-dark-400);
}

.FooterTop-logo {
  grid-column: 2 / span 2;
}

.FooterTop-logoDescription {
  font-size: 12px;
  line-height: 22px;
  color: var(--clr-text-tertiary);
}

.FooterTop-misc {
  grid-column: 9 / span 2;
}

.FooterTop-miscApp {
  display: flex;
  flex-direction: column;
}

.FooterTop-miscAppButton {
  display: flex;
  justify-content: space-between;
}

.FooterTop-miscLanguageTitle,
.FooterTop-miscAppTitle {
  font-size: 14px;
}

.FooterTop-miscAppDescription {
  font-size: 12px;
  color: var(--clr-text-tertiary);
}

.FooterBottom {
  grid-column: 1 / -1;
  align-items: center;

  padding: 32px 0 32px 0;
  background: var(--clr-dark-600);
}

.FooterBottom-copyright {
  grid-column: 2 / span 6;
}

.FooterBottom-copyrightTitle {
  font-size: 18px;
  color: var(--clr-text-primary);
}

.FooterBottom-copyrightDescription {
  font-size: 12px;
  color: var(--clr-text-tertiary);
}

.FooterBottom-discord {
  display: flex;
  grid-column: 10;
  justify-content: flex-end;
}
