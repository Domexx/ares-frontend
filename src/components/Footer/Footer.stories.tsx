import { Story } from '@storybook/react/types-6-0';
import React, { ComponentProps } from 'react';
import FooterComponent, { Props } from './Footer';

import FooterMock from './Footer.mock.json';

const Component = {
  title: 'Components/Footer',
  component: FooterComponent,
  argTypes: {},
};

const Template: Story<ComponentProps<typeof FooterComponent>> = (args) => {
  return <FooterComponent {...args} />;
};

export const Footer = Template.bind({});
export default Component;

const items = FooterMock;

// https://storybook.js.org/docs/react/essentials/controls
Footer.args = {
  items,
  description:
    'Lorem ipsum dolor sit amet, consetetur sadipscing elitr, sed diam nonumy eirmod tempor invidunt ut labore et dolore magna aliquyam erat, sed diam ',
  discord: 'https://discord.gg/uAmqc64S',
  copyright: {
    title: '© 2022 HabboSO is powered by soCMS',
    description: 'Das soCMS wurde speziell für HabboSO entwickelt.',
  },
} as Props;
