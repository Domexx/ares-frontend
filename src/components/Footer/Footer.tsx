import React from 'react';
import Logo from '../../elements/Logo';
import s from './Footer.module.css';

import { ReactComponent as Discord } from '../../assets/static/images/Discord.svg';
import { ReactComponent as DownloadIcon } from '../../assets/static/images/Download.svg';

import { FooterListItem } from '../../shared/interfaces/FooterItem';

import List from './List';
import Button from '../../elements/Button';

export type Copyright = {
  title: string;
  description: string;
};

export type Props = {
  items: FooterListItem[];
  description: string;
  discord: string;
  copyright: Copyright;
};

const Footer: React.FC<Props> = ({ items, description, discord, copyright }) => {
  return (
    <footer className={`${s['Footer']} u-grid`}>
      <div className={`${s['FooterTop']} u-grid`}>
        <div className={s['FooterTop-logo']}>
          <Logo />
          <p className={s['FooterTop-logoDescription']}>{description}</p>
        </div>
        <List items={items} />
        <div className={s['FooterTop-misc']}>
          <div className={s['FooterTop-miscLanguage']}>
            <span className={s['FooterTop-miscLanguageTitle']}>Sprachauswahl</span>
          </div>
          <div className={s['FooterTop-miscApp']}>
            <span className={s['FooterTop-miscAppTitle']}>Desktop App</span>
            <p className={s['FooterTop-miscAppDescription']}>
              Downloade dir die Desktop App, um einfacher ins Hotel zukommen.
            </p>
            <Button>
              <div className={s['FooterTop-miscAppButton']}>
                <span>HabboSO downloaden</span>
                <DownloadIcon />
              </div>
            </Button>
          </div>
        </div>
      </div>
      <div className={`${s['FooterBottom']} u-grid`}>
        <div className={s['FooterBottom-copyright']}>
          <div className={s['FooterBottom-copyrightTitle']}>{copyright.title}</div>
          <div className={s['FooterBottom-copyrightDescription']}>{copyright.description}</div>
        </div>
        <a href={discord} className={s['FooterBottom-discord']}>
          <Discord />
        </a>
      </div>
    </footer>
  );
};

export default Footer;
