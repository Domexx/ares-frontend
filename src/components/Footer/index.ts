import loadable from '@loadable/component';

export default loadable(() => {
  return import('./Footer');
});
