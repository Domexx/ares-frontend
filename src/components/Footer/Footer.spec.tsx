import React from 'react';

import { render } from '@testing-library/react';

import Footer from './Footer';

describe('Footer Component', () => {
  it('should render properly', () => {
    const { container } = render(
      <Footer
        items={[]}
        description="Test Description"
        discord="https://discordapp.com"
        copyright={{
          title: 'Copyright Title',
          description: 'Copyright Description',
        }}
      />,
    );

    expect(container.firstChild).toMatchSnapshot();
  });
});
