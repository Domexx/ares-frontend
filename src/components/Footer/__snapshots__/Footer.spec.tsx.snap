// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer Component should render properly 1`] = `
<footer
  class="Footer u-grid"
>
  <div
    class="FooterTop u-grid"
  >
    <div
      class="FooterTop-logo"
    >
      <p
        class="FooterTop-logoDescription"
      >
        Test Description
      </p>
    </div>
    <div
      class="FooterTop-misc"
    >
      <div
        class="FooterTop-miscLanguage"
      >
        <span
          class="FooterTop-miscLanguageTitle"
        >
          Sprachauswahl
        </span>
      </div>
      <div
        class="FooterTop-miscApp"
      >
        <span
          class="FooterTop-miscAppTitle"
        >
          Desktop App
        </span>
        <p
          class="FooterTop-miscAppDescription"
        >
          Downloade dir die Desktop App, um einfacher ins Hotel zukommen.
        </p>
      </div>
    </div>
  </div>
  <div
    class="FooterBottom u-grid"
  >
    <div
      class="FooterBottom-copyright"
    >
      <div
        class="FooterBottom-copyrightTitle"
      >
        Copyright Title
      </div>
      <div
        class="FooterBottom-copyrightDescription"
      >
        Copyright Description
      </div>
    </div>
    <a
      class="FooterBottom-discord"
      href="https://discordapp.com"
    >
      <svg />
    </a>
  </div>
</footer>
`;
